src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpTokenizer.cs

```diff
@@ -547,10 +547,11 @@ private StateResult VerbatimStringLiteral()
             else if (EndOfFile)
             {
                 CurrentErrors.Add(
-                    new RazorError(
-                        LegacyResources.ParseError_Unterminated_String_Literal,
-                        CurrentStart,
-                        length: 1 /* end of file */));
+                    RazorDiagnostic.Create(
```

Comment:
Shouldn't these be created from RazorDiagnosticFactory? Same question elsewhere

Reply:
Yes they should be. I will send a separate PR for that because it is viral and needs a lot more to change for the tests to build

Reply:
kk